homepage/tests/appUser_model_tests.py

```diff
@@ -0,0 +1,56 @@
+import pytest
+from homepage.models import AppUser
+
+
+@pytest.mark.django_db
+class TestAppUser:
+
+    @pytest.fixture
+    def appUsers(self):
+        appUser1 = AppUser.create_appUser('user1', 'user1@mta.ac.il', 'useruser222')
+        appUser2 = AppUser.create_appUser('user2', 'user2@mta.ac.il', '1234')
+        appUser3 = AppUser.create_appUser('user3', 'user3@gmail.com', '10101010')
+
+        return (appUser1, appUser2, appUser3)
+
+    def test_create_appUser(self, appUsers):
+
+        # 1. The 3 appUsers which are created in the fixture above are the correct type of object (AppUser)
+        # 2. The 3 appUsers which are created in the fixture above were sucesfully saved in the DB
+        for appUser in appUsers:
+            assert isinstance(appUser, AppUser)
+            assert AppUser.objects.filter(pk=appUser.pk).exists()
+
+        # 3. An appUser with invalid username is not created
+        try:
+            AppUser.create_appUser('', 'user4@mta.ac.il', 'useruser222')
+        except ValueError:
+            invalid_username_created = False
+
+        assert invalid_username_created is False
+
+        # 4. an appUser with empty password does not have a usuable password (Django built in create_user behaviour)
+        appUser_invalid_password = AppUser.create_appUser('user6', 'user6@mta.ac.il', None)
+        assert (appUser_invalid_password.user).has_usable_password() is False
+
+    def test_toggle_user_activation(self, appUsers):
+        assert appUsers[0].user.is_active is True
+
+        appUsers[0].toggle_user_activation()
+        assert appUsers[0].user.is_active is False
+
+        appUsers[0].toggle_user_activation()
+        assert appUsers[0].user.is_active is True
+
+    def test_get_all_appUsers(self, appUsers):
+        all_AppUsers = AppUser.get_all_appUsers()
+
+        for appUser in appUsers:
+            assert appUser in all_AppUsers
+
+    def test_find_appUser(self, appUsers):
+        assert (AppUser.find_appUser('user1')
+                and AppUser.find_appUser('user2')
+                and AppUser.find_appUser('user3'))
+
+        assert AppUser.find_appUser('unknown_user') is None

```

Comment:
please split it into 2 tests:
1. test_get_app_user (you can do it with only one like 'user2')
2. test_get_undefined_app_user

Reply:
changed